example/advection/advection_driver.cpp

```diff
@@ -44,106 +47,139 @@ AdvectionDriver::AdvectionDriver(ParameterInput *pin, ApplicationInput *app_in,
+  // could still be executed in parallel
+  TaskRegion &single_tasklist_per_pack_region = tc.AddRegion(num_partitions);
+  for (int i = 0; i < num_partitions; i++) {
+    auto &tl = single_tasklist_per_pack_region[i];
+    auto &mbase = pmesh->mesh_data.GetOrAdd("base", i);
+    auto &mc0 = pmesh->mesh_data.GetOrAdd(stage_name[stage - 1], i);
+    auto &mc1 = pmesh->mesh_data.GetOrAdd(stage_name[stage], i);
+    auto &mdudt = pmesh->mesh_data.GetOrAdd("dUdt", i);
+
+    // compute the divergence of fluxes of conserved variables
+    auto flux_div = tl.AddTask(none, parthenon::Update::FluxDivergenceMesh, mc0, mdudt);
+
+    // apply du/dt to all independent fields in the container
+    auto update =
+        tl.AddTask(flux_div, UpdateMeshData, stage, integrator, mc0, mbase, mdudt, mc1);
+  }
+  TaskRegion &async_region2 = tc.AddRegion(num_task_lists_executed_independently);
+
+  for (int i = 0; i < blocks.size(); i++) {
+    auto &pmb = blocks[i];
+    auto &tl = async_region2[i];
+    auto &sc1 = pmb->meshblock_data.Get(stage_name[stage]);
+
+    auto prev_task = none;
+    // update ghost cells
+    auto send = tl.AddTask(none, &MeshBlockData<Real>::SendBoundaryBuffers, sc1.get());
+    auto recv = tl.AddTask(send, &MeshBlockData<Real>::ReceiveBoundaryBuffers, sc1.get());
+    auto fill_from_bufs =
+        tl.AddTask(recv, &MeshBlockData<Real>::SetBoundaries, sc1.get());
+    prev_task = fill_from_bufs;
+
+    auto clear_comm_flags = tl.AddTask(prev_task, &MeshBlockData<Real>::ClearBoundary,
+                                       sc1.get(), BoundaryCommSubset::all);
+
+    auto prolongBound = tl.AddTask(
+        prev_task,
+        [](std::shared_ptr<MeshBlock> pmb) {
+          pmb->pbval->ProlongateBoundaries(0.0, 0.0);
           return TaskStatus::complete;
         },
-        sc1);
+        pmb);
+
+    // set physical boundaries
+    auto set_bc = tl.AddTask(prolongBound, parthenon::ApplyBoundaryConditions, sc1);
+
+    // fill in derived fields
+    auto fill_derived =
+        tl.AddTask(set_bc, parthenon::FillDerivedVariables::FillDerived, sc1);
 
-    // Update refinement
-    if (pmesh->adaptive) {
-      auto tag_refine = tl.AddTask(
+    // estimate next time step
+    if (stage == integrator->nstages) {
+      auto new_dt = tl.AddTask(
           fill_derived,
-          [](MeshBlock *pmb) {
-            pmb->pmr->CheckRefinementCondition();
+          [](std::shared_ptr<MeshBlockData<Real>> &rc) {
+            auto pmb = rc->GetBlockPointer();
+            pmb->SetBlockTimestep(parthenon::Update::EstimateTimestep(rc));
             return TaskStatus::complete;
           },
-          pmb);
+          sc1);
+
+      // Update refinement
+      if (pmesh->adaptive) {
+        auto tag_refine = tl.AddTask(
+            fill_derived,
+            [](std::shared_ptr<MeshBlock> pmb) {
+              pmb->pmr->CheckRefinementCondition();
+              return TaskStatus::complete;
+            },
+            pmb);
+      }
```

Comment:
In a future MR, we should try to make these work over the whole `MeshBlockPack` as there is probably some advantage to doing so.

I think that's a topic for a future MR, however.

Reply:
Yes, that's another item on the list of things to be addressed in a separate MR (sooner than later) as it'll make reviewing easier.